src/Mono.Android/Xamarin.Android.Net/AndroidClientHandler.cs

```diff
@@ -272,10 +277,10 @@ string EncodeUrl (Uri url)
 				if(Logger.LogNet)
 					Logger.Log (LogLevel.Info, LOG_APP, " cancelled");
 
-				httpConnection.Disconnect ();
+				await DisconnectAsync (httpConnection);
 				cancellationToken.ThrowIfCancellationRequested ();
 			}
-			cancellationToken.Register (httpConnection.Disconnect);
+			cancellationToken.Register (async () => await DisconnectAsync (httpConnection));
```

Comment:
Also, it may make sense to unregister the cancellation token when your cancellable work has completed.

```
CancellationTokenRegistration cancelRegistration = cancellationToken.Register (() => whatever);
// Do work....
// lots of work...
cancelRegistration.Dispose ();
````
If the CancellationToken is long-lived it prevents you from accidentally keeping things alive longer than necessary.